### PR TITLE
Zeb/clean clean week 3

### PR DESCRIPTION
**Pull request by zohebaziz**
its so clean

### CODE REVIEW

File: app/src/main/java/jayp/bge/playboikotlin/ZebActivity.kt [hunk 1]
```diff
@@ -1,11 +0,0 @@
...
-
-import androidx.appcompat.app.AppCompatActivity
-import android.os.Bundle
-
-class ZebActivity : AppCompatActivity() {
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_zeb)
-    }
-}
```
Comment (robert-chatterton) [inline app/src/main/java/jayp/bge/playboikotlin/ZebActivity.kt:11]: WASTED!!!

File: app/src/main/java/jayp/bge/playboikotlin/zeb/ZebLoginActivity.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityZebLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        viewModel.zebViewStateLiveData.observe(this) {
+
+            if(it.validLogin) {
```
Comment (robert-chatterton) [inline app/src/main/java/jayp/bge/playboikotlin/zeb/ZebLoginActivity.kt:24]: wow

File: app/src/main/java/jayp/bge/playboikotlin/zeb/ZebViewModel.kt [hunk 3]
```diff
@@ -33,6 +33,14 @@ class ZebViewModel : ViewModel() {
         }
     }
 
+    fun onLoginButtonPress(username: String, password: String) {
+
+        if (username == "zohebaziz" && password == "password") {
```
Comment (robert-chatterton) [inline app/src/main/java/jayp/bge/playboikotlin/zeb/ZebViewModel.kt:38]: bro you're giving away the creds man

File: app/src/main/java/jayp/bge/playboikotlin/zeb/fragments/hurricankotalina/HurricaneKotalinaFragment.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import jayp.bge.playboikotlin.databinding.FragmentHurricaneKotalinaBinding
+
+class HurricaneKotalinaFragment: Fragment() {
```
Comment (robert-chatterton) [inline app/src/main/java/jayp/bge/playboikotlin/zeb/fragments/hurricankotalina/HurricaneKotalinaFragment.kt:11]: this probably sucked to name stuff

File: app/src/main/res/layout/activity_zeb_login.xml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        android:id="@+id/loginButton"
+        android:text="login"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="172dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.498"
+        app:layout_constraintStart_toStartOf="parent" />
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (robert-chatterton) [inline app/src/main/res/layout/activity_zeb_login.xml:62]: xml is so based

File: app/src/main/res/layout/fragment_hurricane_kotalina.xml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <Button
+        android:id="@+id/hurricaneButton"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
```
Comment (robert-chatterton) [inline app/src/main/res/layout/fragment_hurricane_kotalina.xml:23]: mommy? sorry mommy? sorry mommy?

**Review by robert-chatterton — COMMENTED** [review]
looks `good`

File: app/src/main/java/jayp/bge/playboikotlin/zeb/ZebViewModel.kt (hunk 3 above)
Comment (zohebaziz) [reply app/src/main/java/jayp/bge/playboikotlin/zeb/ZebViewModel.kt:38]: free

File: app/src/main/res/layout/fragment_kotlin_ripper.xml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+    <Button
+        android:id="@+id/ripperButton"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/ripperText"
+        android:text="rip"/>
```
Comment (pelusodan) [inline app/src/main/res/layout/fragment_kotlin_ripper.xml:24]: no extracting as a string resource? wtf?????????????

File: app/src/main/res/menu/zeb_nav.xml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+    <item
+        android:id="@+id/kotlin_ripper"
+        android:icon="@drawable/ic_action_name"
+        android:title="Kotlin Ripper"/>
+
+    <item
+        android:id="@+id/hurricane_kotalina"
+        android:icon="@drawable/ic_kotalina"
+        android:title="Hurrican Kotalina"/>
```
Comment (pelusodan) [inline app/src/main/res/menu/zeb_nav.xml:12]:
```suggestion
        android:title="Hurricane Kotalina"/>
```
nice typo, 🤡

File: app/src/main/java/jayp/bge/playboikotlin/zeb/fragments/hurricankotalina/HurricaneKotalinaFragment.kt [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+    private lateinit var binding: FragmentHurricaneKotalinaBinding
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
```
Comment (pelusodan) [inline app/src/main/java/jayp/bge/playboikotlin/zeb/fragments/hurricankotalina/HurricaneKotalinaFragment.kt:19]:
```suggestion
    ): View {
```

File: app/src/main/java/jayp/bge/playboikotlin/zeb/fragments/hurricankotalina/HurricaneKotalinaSubFragment.kt [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.navArgs
+import jayp.bge.playboikotlin.databinding.FragmentHurricaneKotalinaSubBinding
+
+class HurricaneKotalinaSubFragment: Fragment() {
```
Comment (pelusodan) [inline app/src/main/java/jayp/bge/playboikotlin/zeb/fragments/hurricankotalina/HurricaneKotalinaSubFragment.kt:11]: what like this is a sub? like its your sub? what is this? bdsm???

File: app/src/main/java/jayp/bge/playboikotlin/zeb/fragments/kotlinripper/KotlinRipperFragment.kt [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+class KotlinRipperFragment : Fragment() {
+
+    private lateinit var binding: FragmentKotlinRipperBinding
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
```
Comment (pelusodan) [inline app/src/main/java/jayp/bge/playboikotlin/zeb/fragments/kotlinripper/KotlinRipperFragment.kt:18]:
```suggestion
    ): View {
```

File: app/src/main/res/layout/activity_zeb.xml [hunk 12]
```diff
@@ -57,6 +57,16 @@
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
+    <Button
+        android:id="@+id/bottomestButton"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="bottomest button"
```
Comment (pelusodan) [inline app/src/main/res/layout/activity_zeb.xml:64]: extract as a string resource

File: app/src/main/res/layout/activity_zeb_login.xml [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+
+    <Button
+        android:id="@+id/loginButton"
+        android:text="login"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="172dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.498"
```
Comment (pelusodan) [inline app/src/main/res/layout/activity_zeb_login.xml:60]: remove the horizontal bias

**Review by pelusodan — APPROVED** [review]

**Pull request merged**